sources/packages/api/src/route-controllers/student/student.aest.controller.ts

```diff
@@ -40,6 +77,105 @@ export class StudentAESTController extends BaseController {
+      response,
+      uniqueFileName,
+    );
+  }
+
+  /**
+   * Allow files uploads to a particular student.
+   * @param userToken authentication token.
+   * @param studentId student to receive the uploaded file.
+   * @param file file content.
+   * @param uniqueFileName unique file name (name+guid).
+   * @param groupName friendly name to group files. Currently using
+   * the value from 'Directory' property from form.IO file component.
+   * @returns created file information.
+   */
+  @Post(":studentId/files")
+  @ApiNotFoundResponse({ description: "Student was not found." })
+  @UseInterceptors(
+    FileInterceptor("file", {
+      limits: uploadLimits(MAX_UPLOAD_FILES, MAX_UPLOAD_PARTS),
+      fileFilter: defaultFileFilter,
+    }),
+  )
+  async uploadFile(
+    @UserToken() userToken: IUserToken,
+    @Param("studentId") studentId: number,
+    @UploadedFile() file: Express.Multer.File,
+    @Body("uniqueFileName") uniqueFileName: string,
+    @Body("group") groupName: string,
+  ): Promise<FileCreateAPIOutDTO> {
+    const studentExists = this.studentService.studentExists(studentId);
+    if (!studentExists) {
+      throw new NotFoundException("Student was not found.");
+    }
+
+    return this.studentControllerService.uploadFile(
+      studentId,
+      file,
+      uniqueFileName,
+      groupName,
+      userToken.userId,
+    );
+  }
+
+  /**
+   * Saves the files submitted by the Ministry to the student.
+   * All the files uploaded are first saved as temporary file in
+   * the DB. When this endpoint is called, the temporary
+   * files (saved during the upload) are updated to its proper
+   * group and file origin.
+   * @param userToken user authentication.
+   * @param studentId student to have the file saved.
+   * @param payload list of files to be saved.
+   */
+  @Patch(":studentId/save-uploaded-files")
+  @ApiNotFoundResponse({ description: "Student not found." })
+  async saveStudentUploadedFiles(
+    @UserToken() userToken: IUserToken,
+    @Param("studentId") studentId: number,
+    @Body() payload: AESTFileUploadToStudentAPIInDTO,
+  ): Promise<void> {
+    const student = await this.studentService.getStudentById(studentId);
+    if (!studentId) {
+      throw new NotFoundException("Student not found.");
+    }
+
+    // This method will be executed alongside with the transaction during the
+    // execution of the method updateStudentFiles.
+    const sendFileUploadNotification = () =>
+      this.gcNotifyActionsService.sendMinistryFileUploadNotification({
+        firstName: student.user.firstName,
+        lastName: student.user.lastName,
+        toAddress: student.user.email,
+      });
+    // Updates the previously temporary uploaded files.
+    await this.fileService.updateStudentFiles(
+      studentId,
+      userToken.userId,
+      payload.associatedFiles,
+      FileOriginType.Ministry,
+      MINISTRY_FILE_UPLOAD_GROUP_NAME,
+      undefined,
```

Comment:
@andrewsignori-aot , if we change the parameters position, we can remove undefined. Also declare the sendFileUploadNotification as default boolean false in the other calling methods.